mobify/progressive-web-scaffold
# Automatically convert Immutable objects in selectors

This PR brings in the `reselect-immutable-helpers` library (which is the open source version of `selector-utils.js`) and uses its newest features to avoid having to wrap only some selectors in `selectorToJS`.

We now have a function `createPropsSelector` that wraps `createStructuredSelector`, wrapping each of the input selectors in `ensureJSSelector`, a function that converts Immutable objects to plain JS but still works if the object isn't Immutable. We can't tell ahead of time from the selector the type of the object, but we can tell when the selector is run. Source for these functions is at https://github.com/DanielBrookRoberge/reselect-immutable-helpers/blob/master/index.js#L31

 **JIRA**: (link to JIRA ticket)
 **Linked PRs**: (links to corresponding PRs, optional)

## Changes
- Convert from `createStructuredSelector` to `createPropsSelector`
- Remove now-unnecessary uses of `selectorToJS`
- Remove duplicate `selectorToJS` definition from `selector-utils.js`

## How to test-drive this PR
- (necessary config changes)
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- make sure everything still works
